### ISSUE TITLE
[v18.49.3-3.0.1-(105) Latest] Crash on video view

### ISSUE THREAD

**Issue by SandiyosDev** [therealFoxster/uYouPlus#2]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Have you read the FAQ?

- [X] Yes, I have read the FAQ

### Device information

- iOS/iPadOS version:  iOS 16.1.2, iPadOS 15.2
- Device model: iPhone Xr, iPad Air (3rd Gen)
- Sideload tool (AltStore, Sideloadly, TrollStore, etc.):  TrollStore

### Describe the issue

Crash on video view, _EXC_CRASH (SIGABRT)_ caused by _abort() called_ caused by _VM - pmap_enter retried due to resource shortage_, the same crash was reproducible on two distinct devices.

### Steps to reproduce

1. Click on a video 
2. Crash

### Crash log

```shell
Incident Identifier: (redacted)
CrashReporter Key:   (redacted)
Hardware Model:      iPhone11,8
Process:             YouTube [4632]
Path:                /private/var/containers/Bundle/Application/128DCB53-9DE7-4F94-82B6-BC145A1F623B/YouTube.app/YouTube
Identifier:          com.google.ios.youtube
Version:             18.49.3 (18.49.3)
AppStoreTools:       15C5065a
Code Type:           ARM-64 (Native)
Role:                Foreground
Parent Process:      launchd [1]
Coalition:           com.google.ios.youtube [728]

Date/Time:           2024-01-16 14:06:59.4019 +0800
Launch Time:         2024-01-16 14:06:45.0205 +0800
OS Version:          iPhone OS 16.1.2 (20B110)
Release Type:        User
Baseband Version:    5.01.01
Report Version:      104

Exception Type:  EXC_CRASH (SIGABRT)
Exception Codes: 0x0000000000000000, 0x0000000000000000
Triggered by Thread:  0

Application Specific Information:
abort() called


Last Exception Backtrace:
0   CoreFoundation                	       0x1c4d29e88 __exceptionPreprocess + 164
1   libobjc.A.dylib               	       0x1be0638d8 objc_exception_throw + 60
2   CoreFoundation                	       0x1c4e9e84c +[NSObject(NSObject) _copyDescription] + 0
3   UIKitCore                     	       0x1c7c2c5ec -[UIResponder doesNotRecognizeSelector:] + 260
4   CoreFoundation                	       0x1c4d3ffa0 ___forwarding___ + 1592
5   CoreFoundation                	       0x1c4da8350 _CF_forwarding_prep_0 + 96
6   uYou.dylib                    	       0x10552c564 0x10435c000 + 18679140
7   UIKitCore                     	       0x1c6ee2020 -[UIView(CALayerDelegate) layoutSublayersOfLayer:] + 1980
8   QuartzCore                    	       0x1c63b79ec CA::Layer::layout_if_needed(CA::Transaction*) + 500
9   QuartzCore                    	       0x1c63cb0a0 CA::Layer::layout_and_display_if_needed(CA::Transaction*) + 148
10  QuartzCore                    	       0x1c63dc5b0 CA::Context::commit_transaction(CA::Transaction*, double, double*) + 456
11  QuartzCore                    	       0x1c64115ec CA::Transaction::commit() + 652
12  QuartzCore                    	       0x1c63d724c CA::Display::DisplayLink::dispatch_items(unsigned long long, unsigned long long, unsigned long long) + 1444
13  QuartzCore                    	       0x1c63e88a4 display_timer_callback(__CFMachPort*, void*, long, void*) + 372
14  CoreFoundation                	       0x1c4d9b820 __CFMachPortPerform + 176
15  CoreFoundation                	       0x1c4db8d00 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE1_PERFORM_FUNCTION__ + 60
16  CoreFoundation                	       0x1c4dba908 __CFRunLoopDoSource1 + 520
17  CoreFoundation                	       0x1c4d9c13c __CFRunLoopRun + 2264
18  CoreFoundation                	       0x1c4da0ed4 CFRunLoopRunSpecific + 612
19  GraphicsServices              	       0x1fe0a2368 GSEventRunModal + 164
20  UIKitCore                     	       0x1c727f3d0 -[UIApplication _run] + 888
21  UIKitCore                     	       0x1c727f034 UIApplicationMain + 340
22  YouTube                       	       0x100ad59bc 0x100ad0000 + 22972
23  dyld                          	       0x1e340c960 start + 2528

Kernel Triage:
VM - pmap_enter retried due to resource shortage
VM - pmap_enter retried due to resource shortage
VM - pmap_enter retried due to resource shortage
VM - pmap_enter retried due to resource shortage
VM - pmap_enter retried due to resource shortage


Thread 0 name:   Dispatch queue: com.apple.main-thread
Thread 0 Crashed:
0   libsystem_kernel.dylib        	       0x201928200 __pthread_kill + 8
1   libsystem_pthread.dylib       	       0x211d8a1ac pthread_kill + 268
2   libsystem_c.dylib             	       0x1cc35dc8c abort + 180
3   libc++abi.dylib               	       0x211ccab8c abort_message + 132
4   libc++abi.dylib               	       0x211cbaa80 demangling_terminate_handler() + 336
5   libobjc.A.dylib               	       0x1be069d3c _objc_terminate() + 144
6   libc++abi.dylib               	       0x211cc9f28 std::__terminate(void (*)()) + 20
7   libc++abi.dylib               	       0x211cccecc __cxa_rethrow + 148
8   libobjc.A.dylib               	       0x1be0656b4 objc_exception_rethrow + 44
9   CoreFoundation                	       0x1c4da0f98 CFRunLoopRunSpecific + 808
10  GraphicsServices              	       0x1fe0a2368 GSEventRunModal + 164
11  UIKitCore                     	       0x1c727f3d0 -[UIApplication _run] + 888
12  UIKitCore                     	       0x1c727f034 UIApplicationMain + 340
13  YouTube                       	       0x100ad59bc 0x100ad0000 + 22972
14  dyld                          	       0x1e340c960 start + 2528

Thread 1:
0   libsystem_pthread.dylib       	       0x211d83b90 start_wqthread + 0

Thread 2:
0   libsystem_pthread.dylib       	       0x211d83b90 start_wqthread + 0

Thread 3 name:  com.apple.uikit.eventfetch-thread
Thread 3:
0   libsystem_kernel.dylib        	       0x201921b48 mach_msg2_trap + 8
1   libsystem_kernel.dylib        	       0x201934008 mach_msg2_internal + 80
2   libsystem_kernel.dylib        	       0x201934248 mach_msg_overwrite + 388
3   libsystem_kernel.dylib        	       0x20192208c mach_msg + 24
4   CoreFoundation                	       0x1c4d9aaf0 __CFRunLoopServiceMachPort + 160
5   CoreFoundation                	       0x1c4d9bd34 __CFRunLoopRun + 1232
6   CoreFoundation                	       0x1c4da0ed4 CFRunLoopRunSpecific + 612
7   Foundation                    	       0x1bf14e334 -[NSRunLoop(NSRunLoop) runMode:beforeDate:] + 212
8   Foundation                    	       0x1bf14e21c -[NSRunLoop(NSRunLoop) runUntilDate:] + 64
9   UIKitCore                     	       0x1c73b433c -[UIEventFetcher threadMain] + 436
10  Foundation                    	       0x1bf167808 __NSThread__start__ + 716
11  libsystem_pthread.dylib       	       0x211d846cc _pthread_start + 148
12  libsystem_pthread.dylib       	       0x211d83ba4 thread_start + 8

Thread 4:
0   libsystem_pthread.dylib       	       0x211d83b90 start_wqthread + 0

Thread 5 name:  V8 DefaultWorke
Thread 5:
0   libsystem_kernel.dylib        	       0x20192241c __psynch_cvwait + 8
1   libsystem_pthread.dylib       	       0x211d8b06c _pthread_cond_wait + 1232
2   Module_Framework              	       0x10c91c420 0x10c268000 + 7029792
3   Module_Framework              	       0x10c91c2d4 0x10c268000 + 7029460
4   Module_Framework              	       0x10c91c29c 0x10c268000 + 7029404
5   libsystem_pthread.dylib       	       0x211d846cc _pthread_start + 148
6   libsystem_pthread.dylib       	       0x211d83ba4 thread_start + 8

Thread 6 name:  V8 DefaultWorke
Thread 6:
0   libsystem_kernel.dylib        	       0x20192241c __psynch_cvwait + 8
1   libsystem_pthread.dylib       	       0x211d8b06c _pthread_cond_wait + 1232
2   Module_Framework              	       0x10c91c420 0x10c268000 + 7029792
3   Module_Framework              	       0x10c91c2d4 0x10c268000 + 7029460
4   Module_Framework              	       0x10c91c29c 0x10c268000 + 7029404
5   libsystem_pthread.dylib       	       0x211d846cc _pthread_start + 148
6   libsystem_pthread.dylib       	       0x211d83ba4 thread_start + 8

Thread 7 name:  V8 DefaultWorke
Thread 7:
0   libsystem_kernel.dylib        	       0x20192241c __psynch_cvwait + 8
1   libsystem_pthread.dylib       	       0x211d8b06c _pthread_cond_wait + 1232
2   Module_Framework              	       0x10c91c420 0x10c268000 + 7029792
3   Module_Framework              	       0x10c91c2d4 0x10c268000 + 7029460
4   Module_Framework              	       0x10c91c29c 0x10c268000 + 7029404
5   libsystem_pthread.dylib       	       0x211d846cc _pthread_start + 148
6   libsystem_pthread.dylib       	       0x211d83ba4 thread_start + 8

Thread 8 name:  V8 DefaultWorke
Thread 8:
0   libsystem_kernel.dylib        	       0x20192241c __psynch_cvwait + 8
1   libsystem_pthread.dylib       	       0x211d8b06c _pthread_cond_wait + 1232
2   Module_Framework              	       0x10c91c420 0x10c268000 + 7029792
3   Module_Framework              	       0x10c91c2d4 0x10c268000 + 7029460
4   Module_Framework              	       0x10c91c29c 0x10c268000 + 7029404
5   libsystem_pthread.dylib       	       0x211d846cc _pthread_start + 148
6   libsystem_pthread.dylib       	       0x211d83ba4 thread_start + 8

Thread 9 name:  V8 DefaultWorke
Thread 9:
0   libsystem_kernel.dylib        	       0x20192241c __psynch_cvwait + 8
1   libsystem_pthread.dylib       	       0x211d8b06c _pthread_cond_wait + 1232
2   Module_Framework              	       0x10c91c420 0x10c268000 + 7029792
3   Module_Framework              	       0x10c91c2d4 0x10c268000 + 7029460
4   Module_Framework              	       0x10c91c29c 0x10c268000 + 7029404
5   libsystem_pthread.dylib       	       0x211d846cc _pthread_start + 148
6   libsystem_pthread.dylib       	       0x211d83ba4 thread_start + 8

Thread 10:
0   libsystem_pthread.dylib       	       0x211d83b90 start_wqthread + 0

Thread 11:
0   libsystem_pthread.dylib       	       0x211d83b90 start_wqthread + 0

Thread 12 name:  com.apple.NSURLConnectionLoader
Thread 12:
0   libsystem_kernel.dylib        	       0x201921b48 mach_msg2_trap + 8
1   libsystem_kernel.dylib        	       0x201934008 mach_msg2_internal + 80
2   libsystem_kernel.dylib        	       0x201934248 mach_msg_overwrite + 388
3   libsystem_kernel.dylib        	       0x20192208c mach_msg + 24
4   CoreFoundation                	       0x1c4d9aaf0 __CFRunLoopServiceMachPort + 160
5   CoreFoundation                	       0x1c4d9bd34 __CFRunLoopRun + 1232
6   CoreFoundation                	       0x1c4da0ed4 CFRunLoopRunSpecific + 612
7   CFNetwork                     	       0x1c60f85a8 0x1c5ea2000 + 2450856
8   Foundation                    	       0x1bf167808 __NSThread__start__ + 716
9   libsystem_pthread.dylib       	       0x211d846cc _pthread_start + 148
10  libsystem_pthread.dylib       	       0x211d83ba4 thread_start + 8

Thread 13:
0   libsystem_pthread.dylib       	       0x211d83b90 start_wqthread + 0

Thread 14 name:  JSInstanceDisposer
Thread 14:
0   libsystem_kernel.dylib        	       0x20192241c __psynch_cvwait + 8
1   libsystem_pthread.dylib       	       0x211d8b06c _pthread_cond_wait + 1232
2   Module_Framework              	       0x10c90dea8 0x10c268000 + 6971048
3   Module_Framework              	       0x10f43c290 0x10c268000 + 52249232
4   Module_Framework              	       0x1119f8cf0 0x10c268000 + 91819248
5   Module_Framework              	       0x1119f9098 0x10c268000 + 91820184
6   Module_Framework              	       0x10c90da78 youtube::client::blocks::runtime::ts::JSInstanceDisposer::Run() + 124
7   Module_Framework              	       0x10c90ce48 0x10c268000 + 6966856
8   libsystem_pthread.dylib       	       0x211d846cc _pthread_start + 148
9   libsystem_pthread.dylib       	       0x211d83ba4 thread_start + 8

Thread 15 name:  com.apple.CFSocket.private
Thread 15:
0   libsystem_kernel.dylib        	       0x2019226ac __select + 8
1   CoreFoundation                	       0x1c4deb6c4 __CFSocketManager + 636
2   libsystem_pthread.dylib       	       0x211d846cc _pthread_start + 148
3   libsystem_pthread.dylib       	       0x211d83ba4 thread_start + 8

Thread 16 name:  ThreadPoolServiceThread
Thread 16:
0   libsystem_kernel.dylib        	       0x201921b48 mach_msg2_trap + 8
1   libsystem_kernel.dylib        	       0x201934008 mach_msg2_internal + 80
2   libsystem_kernel.dylib        	       0x201934248 mach_msg_overwrite + 388
3   libsystem_kernel.dylib        	       0x20192208c mach_msg + 24
4   CoreFoundation                	       0x1c4d9aaf0 __CFRunLoopServiceMachPort + 160
5   CoreFoundation                	       0x1c4d9bd34 __CFRunLoopRun + 1232
6   CoreFoundation                	       0x1c4da0ed4 CFRunLoopRunSpecific + 612
7   Foundation                    	       0x1bf14e334 -[NSRunLoop(NSRunLoop) runMode:beforeDate:] + 212
8   Module_Framework              	       0x10c34c9c8 0x10c268000 + 936392
9   Module_Framework              	       0x10c34b1bc 0x10c268000 + 930236
10  Module_Framework              	       0x10c30251c 0x10c268000 + 632092
11  Module_Framework              	       0x10c2d0760 0x10c268000 + 427872
12  Module_Framework              	       0x10c3255d8 0x10c268000 + 775640
13  Module_Framework              	       0x10c30fec8 0x10c268000 + 687816
14  Module_Framework              	       0x10c325700 0x10c268000 + 775936
15  Module_Framework              	       0x10c345a68 0x10c268000 + 907880
16  libsystem_pthread.dylib       	       0x211d846cc _pthread_start + 148
17  libsystem_pthread.dylib       	       0x211d83ba4 thread_start + 8

Thread 17 name:  ThreadPoolForegroundWorker
Thread 17:
0   libsystem_kernel.dylib        	       0x201921b48 mach_msg2_trap + 8
1   libsystem_kernel.dylib        	       0x201934008 mach_msg2_internal + 80
2   libsystem_kernel.dylib        	       0x201934248 mach_msg_overwrite + 388
3   libsystem_kernel.dylib        	       0x20192208c mach_msg + 24
4   Module_Framework              	       0x10c34ea58 0x10c268000 + 944728
5   Module_Framework              	       0x10c3190fc 0x10c268000 + 725244
6   Module_Framework              	       0x10c319c88 0x10c268000 + 728200
7   Module_Framework              	       0x10c3198fc 0x10c268000 + 727292
8   Module_Framework              	       0x10c3197f0 0x10c268000 + 727024
9   Module_Framework              	       0x10c345a68 0x10c268000 + 907880
10  libsystem_pthread.dylib       	       0x211d846cc _pthread_start + 148
11  libsystem_pthread.dylib       	       0x211d83ba4 thread_start + 8

Thread 18 name:  ThreadPoolBackgroundWorker
Thread 18:
0   libsystem_kernel.dylib        	       0x201921b48 mach_msg2_trap + 8
1   libsystem_kernel.dylib        	       0x201934008 mach_msg2_internal + 80
2   libsystem_kernel.dylib        	       0x201934248 mach_msg_overwrite + 388
3   libsystem_kernel.dylib        	       0x20192208c mach_msg + 24
4   Module_Framework              	       0x10c34ea58 0x10c268000 + 944728
5   Module_Framework              	       0x10c3190fc 0x10c268000 + 725244
6   Module_Framework              	       0x10c319a84 0x10c268000 + 727684
7   Module_Framework              	       0x10c319878 0x10c268000 + 727160
8   Module_Framework              	       0x10c319820 0x10c268000 + 727072
9   Module_Framework              	       0x10c345a68 0x10c268000 + 907880
10  libsystem_pthread.dylib       	       0x211d846cc _pthread_start + 148
11  libsystem_pthread.dylib       	       0x211d83ba4 thread_start + 8

Thread 19 name:  NetworkConfigWatcher
Thread 19:
0   libsystem_kernel.dylib        	       0x201921b48 mach_msg2_trap + 8
1   libsystem_kernel.dylib        	       0x201934008 mach_msg2_internal + 80
2   libsystem_kernel.dylib        	       0x201934248 mach_msg_overwrite + 388
3   libsystem_kernel.dylib        	       0x20192208c mach_msg + 24
4   CoreFoundation                	       0x1c4d9aaf0 __CFRunLoopServiceMachPort + 160
5   CoreFoundation                	       0x1c4d9bd34 __CFRunLoopRun + 1232
6   CoreFoundation                	       0x1c4da0ed4 CFRunLoopRunSpecific + 612
7   Foundation                    	       0x1bf14e334 -[NSRunLoop(NSRunLoop) runMode:beforeDate:] + 212
8   Module_Framework              	       0x10c34c9c8 0x10c268000 + 936392
9   Module_Framework              	       0x10c34b1bc 0x10c268000 + 930236
10  Module_Framework              	       0x10c30251c 0x10c268000 + 632092
11  Module_Framework              	       0x10c2d0760 0x10c268000 + 427872
12  Module_Framework              	       0x10c3255d8 0x10c268000 + 775640
13  Module_Framework              	       0x10c325700 0x10c268000 + 775936
14  Module_Framework              	       0x10c345a68 0x10c268000 + 907880
15  libsystem_pthread.dylib       	       0x211d846cc _pthread_start + 148
16  libsystem_pthread.dylib       	       0x211d83ba4 thread_start + 8

Thread 20 name:  network
Thread 20:
0   libsystem_kernel.dylib        	       0x201921b48 mach_msg2_trap + 8
1   libsystem_kernel.dylib        	       0x201934008 mach_msg2_internal + 80
2   libsystem_kernel.dylib        	       0x201934248 mach_msg_overwrite + 388
3   libsystem_kernel.dylib        	       0x20192208c mach_msg + 24
4   CoreFoundation                	       0x1c4d9aaf0 __CFRunLoopServiceMachPort + 160
5   CoreFoundation                	       0x1c4d9bd34 __CFRunLoopRun + 1232
6   CoreFoundation                	       0x1c4da0ed4 CFRunLoopRunSpecific + 612
7   Foundation                    	       0x1bf14e334 -[NSRunLoop(NSRunLoop) runMode:beforeDate:] + 212
8   Module_Framework              	       0x10c34c9c8 0x10c268000 + 936392
9   Module_Framework              	       0x10c34b1bc 0x10c268000 + 930236
10  Module_Framework              	       0x10c30251c 0x10c268000 + 632092
11  Module_Framework              	       0x10c2d0760 0x10c268000 + 427872
12  Module_Framework              	       0x10c3255d8 0x10c268000 + 775640
13  Module_Framework              	       0x10c325700 0x10c268000 + 775936
14  Module_Framework              	       0x10c345a68 0x10c268000 + 907880
15  libsystem_pthread.dylib       	       0x211d846cc _pthread_start + 148
16  libsystem_pthread.dylib       	       0x211d83ba4 thread_start + 8

Thread 21 name:  Network File Thread
Thread 21:
0   libsystem_kernel.dylib        	       0x201921b48 mach_msg2_trap + 8
1   libsystem_kernel.dylib        	       0x201934008 mach_msg2_internal + 80
2   libsystem_kernel.dylib        	       0x201934248 mach_msg_overwrite + 388
3   libsystem_kernel.dylib        	       0x20192208c mach_msg + 24
4   CoreFoundation                	       0x1c4d9aaf0 __CFRunLoopServiceMachPort + 160
5   CoreFoundation                	       0x1c4d9bd34 __CFRunLoopRun + 1232
6   CoreFoundation                	       0x1c4da0ed4 CFRunLoopRunSpecific + 612
7   Module_Framework              	       0x10c34c628 0x10c268000 + 935464
8   Module_Framework              	       0x10c34b1bc 0x10c268000 + 930236
9   Module_Framework              	       0x10c30251c 0x10c268000 + 632092
10  Module_Framework              	       0x10c2d0760 0x10c268000 + 427872
11  Module_Framework              	       0x10c3255d8 0x10c268000 + 775640
12  Module_Framework              	       0x10c325700 0x10c268000 + 775936
13  Module_Framework              	       0x10c345a68 0x10c268000 + 907880
14  libsystem_pthread.dylib       	       0x211d846cc _pthread_start + 148
15  libsystem_pthread.dylib       	       0x211d83ba4 thread_start + 8

Thread 22 name:  ThreadPoolForegroundWorker
Thread 22:
0   libsystem_kernel.dylib        	       0x201921b48 mach_msg2_trap + 8
1   libsystem_kernel.dylib        	       0x201934008 mach_msg2_internal + 80
2   libsystem_kernel.dylib        	       0x201934248 mach_msg_overwrite + 388
3   libsystem_kernel.dylib        	       0x20192208c mach_msg + 24
4   Module_Framework              	       0x10c34ea58 0x10c268000 + 944728
5   Module_Framework              	       0x10c3190fc 0x10c268000 + 725244
6   Module_Framework              	       0x10c319a84 0x10c268000 + 727684
7   Module_Framework              	       0x10c3198fc 0x10c268000 + 727292
8   Module_Framework              	       0x10c3197f0 0x10c268000 + 727024
9   Module_Framework              	       0x10c345a68 0x10c268000 + 907880
10  libsystem_pthread.dylib       	       0x211d846cc _pthread_start + 148
11  libsystem_pthread.dylib       	       0x211d83ba4 thread_start + 8

Thread 23 name:  com.apple.coremedia.rootQueue.47
Thread 23:
0   libsystem_kernel.dylib        	       0x201921adc semaphore_timedwait_trap + 8
1   libdispatch.dylib             	       0x1cc2fa5f4 _dispatch_sema4_timedwait + 64
2   libdispatch.dylib             	       0x1cc2fabf4 _dispatch_semaphore_wait_slow + 76
3   libdispatch.dylib             	       0x1cc30b840 _dispatch_worker_thread + 324
4   libsystem_pthread.dylib       	       0x211d846cc _pthread_start + 148
5   libsystem_pthread.dylib       	       0x211d83ba4 thread_start + 8

Thread 24 name:  com.apple.coremedia.rootQueue.47
Thread 24:
0   libsystem_kernel.dylib        	       0x201921adc semaphore_timedwait_trap + 8
1   libdispatch.dylib             	       0x1cc2fa5f4 _dispatch_sema4_timedwait + 64
2   libdispatch.dylib             	       0x1cc2fabf4 _dispatch_semaphore_wait_slow + 76
3   libdispatch.dylib             	       0x1cc30b840 _dispatch_worker_thread + 324
4   libsystem_pthread.dylib       	       0x211d846cc _pthread_start + 148
5   libsystem_pthread.dylib       	       0x211d83ba4 thread_start + 8

Thread 25 name:  JavaScriptCore libpas scavenger
Thread 25:
0   libsystem_kernel.dylib        	       0x20192241c __psynch_cvwait + 8
1   libsystem_pthread.dylib       	       0x211d8b06c _pthread_cond_wait + 1232
2   JavaScriptCore                	       0x1d83c23ec scavenger_thread_main + 1364
3   libsystem_pthread.dylib       	       0x211d846cc _pthread_start + 148
4   libsystem_pthread.dylib       	       0x211d83ba4 thread_start + 8

Thread 26 name:  Heap Helper Thread
Thread 26:
0   libsystem_kernel.dylib        	       0x20192241c __psynch_cvwait + 8
1   libsystem_pthread.dylib       	       0x211d8b06c _pthread_cond_wait + 1232
2   JavaScriptCore                	       0x1d83186a4 WTF::ParkingLot::parkConditionallyImpl(void const*, WTF::ScopedLambda<bool ()> const&, WTF::ScopedLambda<void ()> const&, WTF::TimeWithDynamicClockType const&) + 1808
3   JavaScriptCore                	       0x1d82d90fc bool WTF::Condition::waitUntilUnchecked<WTF::Lock>(WTF::Lock&, WTF::TimeWithDynamicClockType const&) + 304
4   JavaScriptCore                	       0x1d82d9578 WTF::Detail::CallableWrapper<WTF::AutomaticThread::start(WTF::AbstractLocker const&)::$_0, void>::call() + 248
5   JavaScriptCore                	       0x1d833ae5c WTF::Thread::entryPoint(WTF::Thread::NewThreadContext*) + 352
6   JavaScriptCore                	       0x1d833d0e4 WTF::wtfThreadEntryPoint(void*) + 16
7   libsystem_pthread.dylib       	       0x211d846cc _pthread_start + 148
8   libsystem_pthread.dylib       	       0x211d83ba4 thread_start + 8

Thread 27 name:  Heap Helper Thread
Thread 27:
0   libsystem_kernel.dylib        	       0x20192241c __psynch_cvwait + 8
1   libsystem_pthread.dylib       	       0x211d8b06c _pthread_cond_wait + 1232
2   JavaScriptCore                	       0x1d83186a4 WTF::ParkingLot::parkConditionallyImpl(void const*, WTF::ScopedLambda<bool ()> const&, WTF::ScopedLambda<void ()> const&, WTF::TimeWithDynamicClockType const&) + 1808
3   JavaScriptCore                	       0x1d82d90fc bool WTF::Condition::waitUntilUnchecked<WTF::Lock>(WTF::Lock&, WTF::TimeWithDynamicClockType const&) + 304
4   JavaScriptCore                	       0x1d82d9578 WTF::Detail::CallableWrapper<WTF::AutomaticThread::start(WTF::AbstractLocker const&)::$_0, void>::call() + 248
5   JavaScriptCore                	       0x1d833ae5c WTF::Thread::entryPoint(WTF::Thread::NewThreadContext*) + 352
6   JavaScriptCore                	       0x1d833d0e4 WTF::wtfThreadEntryPoint(void*) + 16
7   libsystem_pthread.dylib       	       0x211d846cc _pthread_start + 148
8   libsystem_pthread.dylib       	       0x211d83ba4 thread_start + 8

Thread 28 name:  Heap Helper Thread
Thread 28:
0   libsystem_kernel.dylib        	       0x20192241c __psynch_cvwait + 8
1   libsystem_pthread.dylib       	       0x211d8b06c _pthread_cond_wait + 1232
2   JavaScriptCore                	       0x1d83186a4 WTF::ParkingLot::parkConditionallyImpl(void const*, WTF::ScopedLambda<bool ()> const&, WTF::ScopedLambda<void ()> const&, WTF::TimeWithDynamicClockType const&) + 1808
3   JavaScriptCore                	       0x1d82d90fc bool WTF::Condition::waitUntilUnchecked<WTF::Lock>(WTF::Lock&, WTF::TimeWithDynamicClockType const&) + 304
4   JavaScriptCore                	       0x1d82d9578 WTF::Detail::CallableWrapper<WTF::AutomaticThread::start(WTF::AbstractLocker const&)::$_0, void>::call() + 248
5   JavaScriptCore                	       0x1d833ae5c WTF::Thread::entryPoint(WTF::Thread::NewThreadContext*) + 352
6   JavaScriptCore                	       0x1d833d0e4 WTF::wtfThreadEntryPoint(void*) + 16
7   libsystem_pthread.dylib       	       0x211d846cc _pthread_start + 148
8   libsystem_pthread.dylib       	       0x211d83ba4 thread_start + 8

Thread 29 name:  Heap Helper Thread
Thread 29:
0   libsystem_kernel.dylib        	       0x20192241c __psynch_cvwait + 8
1   libsystem_pthread.dylib       	       0x211d8b06c _pthread_cond_wait + 1232
2   JavaScriptCore                	       0x1d83186a4 WTF::ParkingLot::parkConditionallyImpl(void const*, WTF::ScopedLambda<bool ()> const&, WTF::ScopedLambda<void ()> const&, WTF::TimeWithDynamicClockType const&) + 1808
3   JavaScriptCore                	       0x1d82d90fc bool WTF::Condition::waitUntilUnchecked<WTF::Lock>(WTF::Lock&, WTF::TimeWithDynamicClockType const&) + 304
4   JavaScriptCore                	       0x1d82d9578 WTF::Detail::CallableWrapper<WTF::AutomaticThread::start(WTF::AbstractLocker const&)::$_0, void>::call() + 248
5   JavaScriptCore                	       0x1d833ae5c WTF::Thread::entryPoint(WTF::Thread::NewThreadContext*) + 352
6   JavaScriptCore                	       0x1d833d0e4 WTF::wtfThreadEntryPoint(void*) + 16
7   libsystem_pthread.dylib       	       0x211d846cc _pthread_start + 148
8   libsystem_pthread.dylib       	       0x211d83ba4 thread_start + 8

Thread 30 name:  Heap Helper Thread
Thread 30:
0   libsystem_kernel.dylib        	       0x20192241c __psynch_cvwait + 8
1   libsystem_pthread.dylib       	       0x211d8b06c _pthread_cond_wait + 1232
2   JavaScriptCore                	       0x1d83186a4 WTF::ParkingLot::parkConditionallyImpl(void const*, WTF::ScopedLambda<bool ()> const&, WTF::ScopedLambda<void ()> const&, WTF::TimeWithDynamicClockType const&) + 1808
3   JavaScriptCore                	       0x1d82d90fc bool WTF::Condition::waitUntilUnchecked<WTF::Lock>(WTF::Lock&, WTF::TimeWithDynamicClockType const&) + 304
4   JavaScriptCore                	       0x1d82d9578 WTF::Detail::CallableWrapper<WTF::AutomaticThread::start(WTF::AbstractLocker const&)::$_0, void>::call() + 248
5   JavaScriptCore                	       0x1d833ae5c WTF::Thread::entryPoint(WTF::Thread::NewThreadContext*) + 352
6   JavaScriptCore                	       0x1d833d0e4 WTF::wtfThreadEntryPoint(void*) + 16
7   libsystem_pthread.dylib       	       0x211d846cc _pthread_start + 148
8   libsystem_pthread.dylib       	       0x211d83ba4 thread_start + 8

Thread 31:
0   libsystem_pthread.dylib       	       0x211d83b90 start_wqthread + 0

Thread 32:
0   libsystem_pthread.dylib       	       0x211d83b90 start_wqthread + 0

Thread 33:
0   libsystem_pthread.dylib       	       0x211d83b90 start_wqthread + 0

Thread 34:
0   libsystem_pthread.dylib       	       0x211d83b90 start_wqthread + 0

Thread 35:
0   libsystem_pthread.dylib       	       0x211d83b90 start_wqthread + 0

Thread 36:
0   libsystem_pthread.dylib       	       0x211d83b90 start_wqthread + 0


Thread 0 crashed with ARM Thread State (64-bit):
    x0: 0x0000000000000000   x1: 0x0000000000000000   x2: 0x0000000000000000   x3: 0x0000000000000000
    x4: 0x0000000211cce0f5   x5: 0x000000016f32f0d0   x6: 0x000000000000006e   x7: 0x0000000000000000
    x8: 0x7a30782ac4068d14   x9: 0x7a307828df3fd414  x10: 0x0000000000000200  x11: 0x000000000000000b
   x12: 0x000000000000000b  x13: 0x00000000001ff800  x14: 0x00000000000007fb  x15: 0x00000000b9833825
   x16: 0x0000000000000148  x17: 0x000000021b395900  x18: 0x0000000000000000  x19: 0x0000000000000006
   x20: 0x0000000000000103  x21: 0x000000021b3959e0  x22: 0x000000021c790580  x23: 0x0000000281cb45c0
   x24: 0x0000000000000001  x25: 0x0000000000000001  x26: 0x0000000000000001  x27: 0x0000000000000000
   x28: 0x00000001e5cd8980   fp: 0x000000016f32f040   lr: 0x0000000211d8a1ac
    sp: 0x000000016f32f020   pc: 0x0000000201928200 cpsr: 0x40000000
   far: 0x00000002187fbf50  esr: 0x56000080  Address size fault
```

### Are you using the latest version of uYouPlus? If not, why?

✅ Yes, I'm using the latest version of uYouPlus

### Does the issue happen with the official YouTube app from the App Store?

❌ No, I cannot reproduce this issue in the official YouTube app from the App Store

**Comment by therealFoxster** [comment]
Thanks for reporting! Here's some temporary fixes you can try:
- Disabling uYou > Video Player > Playback Speed Controls
- Use the default dark theme (OLED dark theme can cause crashes sometimes). You can change this by going to YouTube Settings > uYouPlus > App appearance > Dark theme > Default

I know this isn't ideal but should stop the crashing for now. Lmk if it helps.

**Comment by 0196071** [comment]
I had the same problem and disabling the playback speed controls fixed the issue

**Comment by therealFoxster** [comment]
Fixed in 449a8a9
